### PR TITLE
feat: API V2 - Local dev only for now

### PR DESCRIPTION
**Pull request by jgadling**
<details open>
  <summary>Argus Stack Details</summary>
  <br />
  <table>
    <tr>
      <th>Name</th>
      <td>jgadling-apiv2</td>
    </tr>
    <tr>
      <th>Short Name</th>
      <td>rare-grub</td>
    </tr>
    <tr>
      <th>Base URL</th>
      <td><a href="https://rare-grub.dev-cryoet.dev.czi.team" target="_blank">https://rare-grub.dev-cryoet.dev.czi.team</a></td>
    </tr>
  </table>
</details>

---

This PR adds a v2 GraphQL API based on the work in the [platformics repo](https://github.com/chanzuckerberg/platformics). For now it includes its own (slightly modified) version of the base platformics codebase, while we wait for several changes to the base library to be merged upstream. We'll eventually pull in the `platformics` library from pypi to avoid maintaining multiple copies.

The schema is in `apiv2/schema/schema.yaml` and the GraphQL API that gets generated is *relatively* similar to the interface presented by Hasura, *except* that all 1:many relationships implement the [Relay connection spec](https://relay.dev/graphql/connections.htm). That means if we have an API V1 query that looks like this:

```
query MyQuery {
  datasets {
    id
    runs {
      id
    }
  }
}
```

It will look like this in API V2:

```
query MyQuery {
  datasets {
    id
    runs {
      edges {
        node {
          id
        }
      }
    }
  }
}
```

Currently this new API isn't integrated into the same local dev environment as the rest of the containers when running `make init` from the root of the repo. I'm hoping to address that in a later PR.

### CODE REVIEW

File: .github/workflows/workflow-argus-docker-build.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        default: '*'
+      branches_ignore:
+        description: 'Branches to skip build on, comma delimited'
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  argus_builder:
+    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@jgadling/argus-patch1
```
Comment (hspitzley-czi) [inline .github/workflows/workflow-argus-docker-build.yaml:27]: Potential fix:
```suggestion
    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@29df95716d12c77632c850baa5ce1235bf757816
```

File: .github/workflows/workflow-argus-docker-build.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        default: '*'
+      branches_ignore:
+        description: 'Branches to skip build on, comma delimited'
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  argus_builder:
+    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@29df95716d12c77632c850baa5ce1235bf757816
```
Comment (hspitzley-czi) [inline .github/workflows/workflow-argus-docker-build.yaml:27]:
```suggestion
    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@816bb25ba8354230da109d3edd23149d94c1c2f5
```

File: .github/workflows/workflow-argus-docker-build.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        default: '*'
+      branches_ignore:
+        description: 'Branches to skip build on, comma delimited'
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  argus_builder:
+    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@816bb25ba8354230da109d3edd23149d94c1c2f5
```
Comment (hspitzley-czi) [inline .github/workflows/workflow-argus-docker-build.yaml:27]:
```suggestion
    uses: chanzuckerberg/github-actions/.github/workflows/argus-docker-build.yaml@ 45068d32dc7313506480e89e8b745c246e6ea42a
```

**Review by manasaV3 — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -19,30 +27,20 @@ repos:
...
-      #    # entry: mypy --config-file packages/backend/pyproject.toml
-      #    # entry: bash -c 'cd packages/backend && mypy --config-file pyproject.toml .'
-      #    additional_dependencies:
-      #      [
-      #        pydantic,
-      #        pydantic-settings,
-      #        click,
-      #      ]
+    - apiv2/pyproject.toml
+    files: '^apiv2/.*'
```
[the remark this answers is not in the document]
Comment (jgadling) [reply .pre-commit-config.yaml:31]: Fair enough. Fixed!

File: .infra/common.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      - name: cerbos-policies
+        emptyDir:
+          sizeLimit: 50Mi
+      - name: keys
+        emptyDir:
+          sizeLimit: 10Mi
+  services:
+    apiv2:
+      image:
+        repository: 533267185808.dkr.ecr.us-west-2.amazonaws.com/core-platform/cryoet-data-portal-backend/apiv2
```
[the remark this answers is not in the document]
Comment (jgadling) [reply .infra/common.yaml:66]: It's just the way the argus team has things set up by default :'( eventually I'd like it to pull from github's image repo, but it doesn't seem worth fighting with right now.

File: .github/workflows/staging-deploy.yaml [hunk 6]
```diff
@@ -5,8 +5,7 @@ on:
     branches:
       - main
     paths:
-      - ".happy/**"
```
[the remark this answers is not in the document]
Comment (jgadling) [reply .github/workflows/staging-deploy.yaml:8]: oops, I didn't mean to change this. Good catch!

File: apiv2/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+name = "platformics"
+version = "0.1.0"
+description = "Platformics bio entities framework"
+authors = ["CZI Infectious Disease Team <help@czid.org>"]
+license = "MIT License"
+readme = "README.md"
+packages = [{include = "platformics"}]
+
+[tool.poetry.scripts]
+platformics = 'platformics.cli.main:cli'
```
[the remark this answers is not in the document]
Comment (jgadling) [reply apiv2/pyproject.toml:11]: ehhh, we're eventually going to be using the upstream `platformics` package and this stuff can go away, but in the meantime it's 🤷‍♀️ ... we can clean it up when we delete the `platformics` directory here?

File: apiv2/graphql_api/schema.graphql [hunk 8]
```diff
@@ -0,0 +1 @@
+Error: No module named 'strawberry.types.field'
```
[the remark this answers is not in the document]
Comment (jgadling) [reply apiv2/graphql_api/schema.graphql:1]: Yikes. fixed!

File: apiv2/database/models/annotation_author.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+from sqlalchemy import Boolean, ForeignKey, Integer, String
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+if TYPE_CHECKING:
+    from database.models.annotation import Annotation
+    from platformics.database.models.file import File
+
+    ...
+else:
+    File = "File"
```
[the remark this answers is not in the document]
Comment (jgadling) [reply apiv2/database/models/annotation_author.py:23]: Nope. I'm not sure why `ruff` isn't removing these unused imports but they're harmless.

File: apiv2/schema/schema.yaml [hunk 10]
```diff
@@ -0,0 +1,2871 @@
...
+    permissible_values:
+      annotation:
+        text: annotation
+        description: The deposition comprises of new annotations for existing datasets
+      dataset:
+        text: dataset
+        description: The deposition comprises of new dataset(s).
+      tomogram:
+        text: tomogram
+        description: The deposition comprises of new tomograms for existing datasets
```
[the remark this answers is not in the document]
Comment (jgadling) [reply apiv2/schema/schema.yaml:301]: We have a 1:many relationship between depositions and this `depositoin_types_enum` via the DepositionTypes type!

**Pull request merged**